nameserver/nameserver.go

```diff
@@ -129,10 +231,13 @@ func (n *Nameserver) ContainerDied(ident string) {
 	entries := n.entries.tombstone(n.ourName, func(e *Entry) bool {
 		if e.ContainerID == ident {
 			n.infof("container %s died; tombstoning entry %s", ident, e.String())
+			// We want to restore all entries if container comes back
+			e.stopped = true
```

Comment:
Yep, because `range` returns a copy of `Entry` which is being modified by the function, and it used to update `n.entries` only if the condition at the linked line holds.